src/vm/methodtable.cpp

```diff
@@ -2275,6 +2278,983 @@ BOOL MethodTable::IsClassPreInited()
+    {
+#ifdef _DEBUG
+        ++fieldNum;
+#endif // _DEBUG
+
+        FieldDesc *pField = pFieldMarshaler->GetFieldDesc();
+        CorElementType fieldType = pField->GetFieldType();
+
+        // Invalid field type.
+        if (fieldType == ELEMENT_TYPE_END)
+        {
+            return false;
+        }
+
+        DWORD fieldOffset = pFieldMarshaler->GetExternalOffset();
+        unsigned normalizedFieldOffset = fieldOffset + startOffsetOfStruct;
+
+        unsigned int fieldNativeSize = pFieldMarshaler->NativeSize();
+        if (fieldNativeSize > SYSTEMV_EIGHT_BYTE_SIZE_IN_BYTES)
+        {
+            // Pass on stack in this case.
+            return false;
+        }
+
+        _ASSERTE(fieldNativeSize != (unsigned int)-1);
+
+        // The field can't span past the end of the struct.
+        if ((normalizedFieldOffset + fieldNativeSize) > helperPtr->structSize)
+        {
+            _ASSERTE(false && "Invalid native struct size. The size of fields and overall size don't agree");
+            return false;
+        }
+
+        SystemVClassificationType fieldClassificationType = SystemVClassificationTypeUnknown;
+
+#ifdef _DEBUG
+        LPCUTF8 fieldName;
+        pField->GetName_NoThrow(&fieldName);
+#endif // _DEBUG
+
+        // Some NStruct Field Types have extra information and require special handling
+        NStructFieldType cls = pFieldMarshaler->GetNStructFieldType();
+        if (cls == NFT_FIXEDCHARARRAYANSI)
+        {
+            fieldClassificationType = SystemVClassificationTypeInteger;
+        }
+        else if (cls == NFT_FIXEDARRAY)
+        {
+            VARTYPE vtElement = ((FieldMarshaler_FixedArray*)pFieldMarshaler)->GetElementVT();
+            switch (vtElement)
+            {
+            case VT_EMPTY:
+                fieldClassificationType = SystemVClassificationTypeInteger;
+                break;
+            case VT_NULL:
+                fieldClassificationType = SystemVClassificationTypeInteger;
+                break;
+            case VT_BOOL:
+                fieldClassificationType = SystemVClassificationTypeInteger;
+                break;
+            case VT_I1:
+                fieldClassificationType = SystemVClassificationTypeInteger;
+                break;
+            case VT_I2:
+                fieldClassificationType = SystemVClassificationTypeInteger;
+                break;
+            case VT_I4:
+                fieldClassificationType = SystemVClassificationTypeInteger;
+                break;
+            case VT_I8:
+                fieldClassificationType = SystemVClassificationTypeInteger;
+                break;
+            case VT_R4:
+                fieldClassificationType = SystemVClassificationTypeSSE;
+                break;
+            case VT_R8:
+                fieldClassificationType = SystemVClassificationTypeSSE;
+                break;
+            case VT_UI1:
+                fieldClassificationType = SystemVClassificationTypeInteger;
```

Comment:
You are missing `break` in multiple cases here. It also looks like it would be more readable if you had one setting of the fieldClassificationType for multiple `case` statements, possibly all that set the same type.